### PR TITLE
rocprim: update to 2.10.4

### PR DESCRIPTION
**Pull request by Wulfsta**

### CODE REVIEW

File: pkgs/development/libraries/hipcub/default.nix [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+    #"${if doCheck then "-DAMDGPU_TEST_TARGETS=${lib.strings.concatStringsSep ";" (config.rocmTargets or ["gfx803" "gfx900" "gfx906"])}" else ""}"
   ];
-  buildInputs = [ hip rocprim ];
-}
+
+  patchPhase = ''
+    sed -e '/find_package(Git/,/endif()/d' \
+        -e '/download_project(/,/^[[:space:]]*)/d' \
+        -i cmake/Dependencies.cmake
+    #sed 's,include(cmake/VerifyCompiler.cmake),,' -i CMakeLists.txt
```
Comment (acowley) [inline pkgs/development/libraries/hipcub/default.nix:38]: Can you delete this line that removes the `VerifyCompiler` inclusion? No need to keep it commented out as it should live on in git history.

Comment (Wulfsta) [reply pkgs/development/libraries/hipcub/default.nix]: This should be fixed now, but I ended up including it in the `hipcub` commit.

**Comment by acowley** [comment]
As [mentioned](https://github.com/Wulfsta/nixos-rocm/commit/cbcd1460ee438a5c3c1b0b718c95aa53c790205e#commitcomment-42709372) in the other thread, I ran the `rocprim` tests in a `nix-shell`, and I ended up with 95% of the tests passing. I don't know if the failures are expected or not, but at least it's not the catastrophic "failed to find device" kind of failure. 100% of the `hipcub` tests pass in `nix-shell`, so that's encouraging!

Let me know if you're ready for this to be merged!

**Comment by Wulfsta** [comment]
I'm fine with this being merged, but I'll continue to think about how we might automate these tests. I think it might be worth it to hunt down what user is building this and see if I can add it to the `video` group to check if that fixes the errors. Being able to automatically test these on a machine with a supported GPU would be good for eventually getting these moved into nixpkgs.

**Pull request merged**